### PR TITLE
AHOYAPPS-473: Decouple remaining Twilio Video SDK dependencies from the RoomActivity

### PR DESCRIPTION
**Pull request by Alton09**
## Description

Decouples the remaining Twilio Video SDK dependencies from the `RoomActivity`. The `RoomManager` now has a `LocalParticipantManager` dependency which is responsible for managing the state of the `LocalParticipant` which is now completely independent of any view lifecycle.

## Validation

- Passed CI
- Manually validated features that are not covered by automated tests

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in ```versionName``` under `app/build.gradle`
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

File: app/src/main/res/menu/room_menu.xml [hunk 1]
```diff
@@ -19,14 +19,6 @@
           android:visible="false"
           app:showAsAction="ifRoom"/>
 
-    <item android:id="@+id/pause_audio_menu_item"
```
Comment (aaalaniz) [inline app/src/main/res/menu/room_menu.xml:22]: Why are these removed? They are handy for QE folks.

**Review by aaalaniz — COMMENTED** [review]
Did a cursory review. The RoomActivity is looking pretty slim.

File: app/src/main/res/menu/room_menu.xml (hunk 1 above)
Comment (Alton09) [reply app/src/main/res/menu/room_menu.xml:22]: Ah yeah, I thought we agreed a while back that it is okay to remove these? Just went ahead and took care of it as part of this PR. Happy to add it back in if needed if QE uses these menu items often. I think these options are confusing to the the average user compared to the FABs that turn off/on video and audio.

Comment (aaalaniz) [reply app/src/main/res/menu/room_menu.xml:22]: they can be useful for debugging differences between unpublishing versus muting/unmuting. Let's add them back. They are not for a typical user, which is why they are in the expanded menu.

File: app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.kt [hunk 2]
```diff
@@ -437,11 +331,6 @@ class RoomActivity : BaseActivity() {
     @OnClick(R.id.connect)
     fun connectButtonClick() {
         InputUtils.hideKeyboard(this)
-        if (!didAcceptPermissions()) {
```
[the remark this answers is not in the document]
Comment (Alton09) [reply app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.kt:440]: This refactor forced me to relook at permission handling so I cleaned it up a bit. I remember talking as a team about this one a while back and how it is a valid use case for a user to be able to connect to a room without video and mic permissions to just listen in on the meeting.

File: app/src/main/java/com/twilio/video/app/sdk/LocalParticipantListener.kt [hunk 3]
```diff
@@ -23,18 +21,7 @@ class LocalParticipantListener(private val roomManager: RoomManager) : LocalPart
         roomManager.sendRoomEvent(NetworkQualityLevelChange(localParticipant.sid, networkQualityLevel))
     }
 
-    override fun onVideoTrackPublished(localParticipant: LocalParticipant, localVideoTrackPublication: LocalVideoTrackPublication) {
```
[the remark this answers is not in the document]
Comment (Alton09) [reply app/src/main/java/com/twilio/video/app/sdk/LocalParticipantListener.kt:26]: So this is no longer needed since any changes to the LocalVideoTrack are communicated via [this line](https://github.com/twilio/twilio-video-app-android/blob/c5fbd93226dd219f9c75d347fa4f2af4c523a131/app/src/main/java/com/twilio/video/app/sdk/LocalParticipantManager.kt#L41) in the `LocalParticipantManager`.

**Review by aaalaniz — APPROVED** [review]
